app/common/state/clearDataState.js

```diff
@@ -0,0 +1,24 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const Immutable = require('immutable')
+const assert = require('assert')
+const { makeImmutable, isMap } = require('./immutableUtil')
+
+const validateState = function (state) {
+  state = makeImmutable(state)
+  assert.ok(isMap(state), 'state must be an Immutable.Map')
+  return state
+}
+
+const api = {
+  getClearDefaults: (state) => {
```

Comment:
this could also be used to refactor these areas
https://github.com/brave/browser-laptop/blob/54915c9655e2efd0a44a40fe43f2da679500add7/app/browser/reducers/historyReducer.js#L44
https://github.com/brave/browser-laptop/blob/54915c9655e2efd0a44a40fe43f2da679500add7/app/browser/menu.js#L718
https://github.com/brave/browser-laptop/blob/54915c9655e2efd0a44a40fe43f2da679500add7/app/ledger.js#L184
https://github.com/brave/browser-laptop/blob/54915c9655e2efd0a44a40fe43f2da679500add7/js/stores/appStore.js#L433

Reply:
I did actually convert them but historyReducer had a failing test so I backed out.  I felt it's better to leave those alone since they're outside the scope of this PR.